src/ascii/interger.cairo

```diff
@@ -0,0 +1,220 @@
+use core::option::OptionTrait;
+use alexandria::data_structures::array_ext::ArrayTraitExt;
+use array::{ArrayTrait, Array};
+use traits::{Into, TryInto, DivRem};
+use zeroable::Zeroable;
+
+trait IntergerToAsciiTrait<T, U> {
+    fn to_ascii(self: T) -> U;
+}
+
+// converts intergers into an array of its individual ascii values
+trait IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252>;
+    fn to_inverse_ascii_array(self: T) -> Array<felt252>;
+}
+
+// converts intergers into an array of its individual ascii values
+// e.g. 123 -> [49, 50, 51]
+impl IntergerToAsciiArrayTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = self.to_inverse_ascii_array();
+        new_arr.reverse();
+        new_arr
+    }
+
+    fn to_inverse_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = ArrayTrait::new();
+        if self <= 9.try_into().unwrap() {
```

Comment:
can't you just use into() ?

Reply:
unfortunately no if i were to bring in a generic impl of `Into<felt252, T>` the trait would not work on uints as they do not have a felt252 -> uint Into trait implementation on them only a try_into trait

Reply:
you have
`impl TInto: Into<T, felt252>`
just do 
`if self.into() <= 9_felt252` ?

Reply:
the `PartialOrd` impl is set to type `T` when i tried that i get this error `Trait has no implementation in context: core::traits::PartialOrd::<core::felt252>`

Reply:
ah yes then keep it as is